reputation.php

```diff
@@ -864,17 +864,51 @@
 
 	if(!empty($post_cache))
 	{
-		$sql = implode(',', $post_cache);
+		$pids = implode(',', $post_cache);
+
+		$sql = array("p.pid IN ({$pids})");
+
+		// get forums user cannot view
+		$unviewable = get_unviewable_forums(true);
+		if($unviewable)
+		{
+			$sql[] = "p.fid NOT IN ({$unviewable})";
+		}
+
+		// get inactive forums
+		$inactive = get_inactive_forums();
+		if($inactive)
+		{
+			$sql[] = "p.fid NOT IN ({$inactive})";
+		}
+
+		if(!$mybb->user['ismoderator'])
```

Comment:
> I don't think it makes the code faster either.

For posts may be not, but threads don't have an index on `visible` so I just added it.